zerver/lib/actions.py

```diff
@@ -1835,9 +1835,13 @@ def build_message_send_dict(
     message.rendered_content_version = markdown_version
     links_for_embed = rendering_result.links_for_preview
 
+    mentioned_user_group_ids = list(rendering_result.mentions_user_group_ids)
+    if UserGroup.objects.filter(id__in=mentioned_user_group_ids, is_system_group=True).exists():
+        raise JsonableError(_("You do not have permission to mention system user group."))
```

Comment:
I think if we were to have an error string here, it should be "You do not have permission to mention this user group" -- that way the string doesn't need to change when we in the future add a "what groups can mention this group" configurable setting.

That said, I think it might be better to just have the `get_possible_mentions_info` logic (or similar) exclude system groups from what groups exist for the purpose of mentioning at this phase (where the intent is for these to be invisible in the UI).